react/ui-components/src/hoc/BoundaryFilter.js

```
@@ -0,0 +1,738 @@
+          updatedOptions[childBoundaryType] = updatedOptions[childBoundaryType] || {};
+
+          updatedOptions[childBoundaryType][value?.code] = updatedOptions[childBoundaryType][value?.code] || [];
+
+          const existingEntries = new Set(
+            updatedOptions[childBoundaryType][value?.code].map(child => child.code)
+          );
+
+
+          const uniqueChildren = children.filter(child => !existingEntries.has(child.code));
+          updatedOptions[childBoundaryType][value?.code] = [
+            ...updatedOptions[childBoundaryType][value?.code],
```

Biome (1.9.4)

[error] Unsafe usage of optional chaining.

If it short-circuits with 'undefined' the evaluation will throw TypeError here:


(lint/correctness/noUnsafeOptionalChaining)